perllib/FixMyStreet/App/Controller/Admin/Reports.pm

```diff
@@ -297,7 +297,8 @@ sub edit : Path('/admin/report_edit') : Args(1) {
         my $old_state = $problem->state;
 
         my %columns = (
-            flagged => $c->get_param('flagged') ? 1 : 0,
+            # Only superusers can flag / unflag a report
+            flagged =>  $c->user->is_superuser ? ( $c->get_param('flagged') ? 1 :  0 ) : $problem->flagged,
```

Comment:
We don't need to set all the columns here, only the ones we want to change, so I think it'd be a bit clearer if flagged wasn't set in this default `%columns` creation (given don't need to for non superusers), but then afterwards somewhere you had a `if (superuser) { $columns{flagged} = $c->get_param('flagged') ? 1 :  0; }` similar to what you've done in Users.pm